### PR TITLE
bug: build fails on mac because there is no /bin/true

### PR DESCRIPTION
**Pull request by ultral**
**How to categorize this PR?**
/bug 

**What this PR does / why we need it**:
it allows to run build on mac. if you run build you can face:
```
./build.sh: line 65: /bin/true: No such file or directory
A83502232@T000cac27a gardenlinux % which -a true                            
true: shell built-in command
/usr/bin/true
A83502232@T000cac27a gardenlinux % sudo ln -s /usr/bin/true /bin/true  
```

**Which issue(s) this PR fixes**:
there are no issues

**Special notes for your reviewer**:

**Release note**:
- bugfix: builds on mac

### CODE REVIEW

**Comment by gyptazy** [comment]
Hey @ultral,

thanks for your PR and the fix. You're right, but I'm struggling how you ran into this one... This means the first statement failed for any reason. While your PR looks fine can you please provide your build command (and/or logs) how you ran into this issue?

Thanks and have a nice weekend,
gyptazy

**Comment by ultral** [comment]
hi @gyptazy 

It's very easy to reproduce:
1. get a mac m1
2. checkout the repo
3. run a build

```
A83502232@T000cac27a gardenlinux_orig % GARDENLINUX_BUILD_CRE=docker ./build.sh  --features khost,kvm,_dev .build/
./build.sh: line 65: /bin/true: No such file or directory
```

it might be interesting:
```
A83502232@T000cac27a gardenlinux_orig % which -a true
true: shell built-in command
/usr/bin/true
A83502232@T000cac27a gardenlinux_orig % echo $SHELL
/bin/zsh
```

**Comment by gyptazy** [comment]
Hey @ultral,

ah, thanks. That was probably my fault, because I never call the `build.sh` directly on macOS. While adding the macOS build  support (amd64+arm64) I always called the `Makefile` instead of calling the `build.sh` directly.

Just gave it a short try and was unable to encounter this by the Makefile:
```
+khost-kvm-dev: build-environment $(SECUREBOOT_CRT)
+       ./build.sh $(BUILD_OPTS) --skip-build --features khost,kvm,_dev $(BUILDDIR) $(VERSION)
+
```
Running:
`make khost-kvm-dev` works without any further issues while calling the `build.sh` directly runs into your mentioned issue.

Your fix works and lgtm. Second opinion from core team (@MrBatschner, @Vincinator,...).

Thanks for your fix!

Cheers,
gyptazy

**Review by Vincinator — APPROVED** [review]
thanks! /lgtm

**Review by gyptazy — APPROVED** [review]

**Pull request merged**
